src/main/java/com/code4ro/legalconsultation/common/controller/GlobalExceptionHandler.java

```
@@ -0,0 +1,61 @@
+package com.code4ro.legalconsultation.common.controller;
+
+import org.springframework.core.Ordered;
+import org.springframework.core.annotation.Order;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import javax.persistence.EntityNotFoundException;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Order(Ordered.HIGHEST_PRECEDENCE)
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(LegalValidationException.class)
+    protected ResponseEntity<Object> handleLegalValidationException(final LegalValidationException ex) {
+        final I18nError error = new I18nError(ex.getI18nKey(), ex.getI8nArguments());
+        return buildResponseEntity(HttpStatus.BAD_REQUEST, Collections.singletonList(error), null, null);
```

My only comment here is that I would use more clearly http codes for marking the type of exception that occured.
As I understand, a legal validation exception should be triggered when the server can map the request parameter, it can extract the data, but the data is not respecting internal validation constraints.
I think a 422 Unprocessable Entity would make it more clear to the consumer what the issue is.

---

i imagined the LegalValidationException as a generic/base class for when we want to throw an error with a single translatable message.  wouldn't it be better to add the status code as an argument as well?